addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -69,6 +69,11 @@ def _compute_department_id(self):
 
     @api.model
     def create(self, values):
+        # If it's a basic user then check if the timesheet is his own.
+        if not self.user_has_groups('hr_timesheet.group_hr_timesheet_approver'):
+            if self.env.user.id != values['user_id']:
```

Comment:
user_id isn't required, even if it has a default value. So ``values['user_id']`` could crash.

You can do something like
``` python
if 'user_id' in values and self.env.user.id != values['user_id'] and not self.user_has_groups('hr_timesheet.group_hr_timesheet_approver'):
    raise AccessError(_("You aren't allowed to access to the timesheet that it's not yours"))
```

On the other hand, the message doesn't say obviously what's wrong. You can raise something like: "Sorry, you are only allowed to create yourself timesheets."